[19.0][MIG] partner_stage_only_confirmed

bosd opened the pull request:
standard migration with odoo-module-migrator /ocabot migration partner_stage_only_confirmed

Depends on #2215 

Build upon #2090 

Depends on #2367 (renames res_partner.state -> stage_state; this module now filters on stage_state).

SilvioC2C committed:
[ADD] partner_stage_only_confirmed

yankinmax committed:
[IMP] partner_stage_only_confirmed: pre-commit auto fixes

yankinmax committed:
[MIG] partner_stage_only_confirmed: Migration to 17.0

mymage committed:
Added translation using Weblate (Italian)

Commit:
[IMP] partner_contact_role: pre-commit execution

bizzappdev committed:
[MIG] partner_stage_only_confirmed: Migration to 18.0

bizzappdev committed:
[ADD] partner_stage_only_confirmed: Added test cases.

bosd committed:
[IMP] partner_stage_only_confirmed: pre-commit auto fixes

bosd committed:
[MIG] partner_stage_only_confirmed: Migration to 19.0

bosd committed:
[DO NOT MERGE TEST REQUIREMENTS]

bosd committed:
[IMP] partner_stage_only_confirmed: add tests

bosd committed:
[IMP] partner_stage_only_confirmed: Improve documentation

bosd committed:
[REF] partner_stage_only_confirmed: remove dead code

BhaveshHeliconia reviewed (commented):
@bosd the dependency has been merged. You can drop the last commit and rebase the PR.

Commit:
[FIX] partner_stage_only_confirmed: use renamed stage_state field

partner_stage now exposes the partner stage value as `stage_state`
(res_partner.`state` was renamed to avoid the AmbiguousColumn collision with
account_move.state in the enterprise VAT report — see #2367). Update the
injected Many2one domain and the tests to filter on `stage_state`.

Depends on #2367

BhaveshHeliconia commented on partner_stage_only_confirmed/models/base.py:
```diff
@@ -0,0 +1,65 @@
+                                    domain = "[('stage_state', '=', 'confirmed')]"
+                                else:
+                                    domain = domain[:-1]
+                                    domain += ", ('stage_state', '=', 'confirmed')]"
```
Use `expression.AND`

bosd replied:
The domain here comes from the view arch as a string and can be a dynamic domain (referencing record fields/context), so it can't be evaluated into a list for `expression.AND` (which operates on domain lists). The string append is equivalent to an implicit AND and stays correct even when the existing domain starts with `|`/`!` operators — e.g. `['|', A, B]` becomes `['|', A, B, ('stage_state', '=', 'confirmed')]` which parses as `(A OR B) AND stage_state`. So I've kept the string-based approach here.

bosd commented:
@BhaveshHeliconia #2367 is still open (not merged yet). The last commit here filters on the renamed `stage_state` field introduced by #2367, so it has to stay until that one lands — dropping it now would break the module. I'll rebase/drop the commit once #2367 is merged.

BhaveshHeliconia commented:
> @BhaveshHeliconia #2367 is still open (not merged yet). The last commit here filters on the renamed `stage_state` field introduced by #2367, so it has to stay until that one lands — dropping it now would break the module. I'll rebase/drop the commit once #2367 is merged.

Okay, got it. Thanks for explaination.
